server/sonar-server/src/main/java/org/sonar/server/computation/AnalysisReportService.java

```diff
@@ -130,9 +133,10 @@ private DefaultIssue setFieldDiffs(DefaultIssue issue, String diffFields, Date a
     return issue;
   }
 
-  private DefaultIssue setComponentId(DefaultIssue issue, ReportComponent component) {
+  private DefaultIssue setComponent(DefaultIssue issue, @Nullable ReportComponent component) {
```

Comment:
In most of the case, the component should not be null. The only case I can think of is in the case of deleted components. In the issues table, the component_id/component_uuid must not be null. Shouldn't we be more defensive ?